src/Symfony/Component/BrowserKit/Cookie.php

```diff
@@ -58,11 +58,11 @@ class Cookie
     public function __construct($name, $value, $expires = null, $path = null, $domain = '', $secure = false, $httponly = true, $encodedValue = false)
     {
         if ($encodedValue) {
-            $this->value = urldecode($value);
+            $this->value = rawurldecode($value);
```

Comment:
if we update decoding, `Cookie::fromString` in HttpFoundation should probably be updated as well, note however `urldecode` works for `rawurlencode` perfectly fine, so im not sure we should change it (given `rawurldecode doesnt work for `urlencode`).